pir/pirclient/client.go

```diff
@@ -0,0 +1,84 @@
+package pirclient
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	"github.com/privacylab/talek/common"
+	"github.com/privacylab/talek/pir/xor"
+)
+
+// Client handles the basic functionalities of a PIR client,
+// generating request vectors and combining partial responses
+type Client struct {
+	log  *common.Logger
+	name string
+}
+
+// NewClient creates a new PIR client
+func NewClient(name string) *Client {
+	c := &Client{}
+	c.log = common.NewLogger(name)
+	c.name = name
+	return c
+}
+
+// GenerateRequestVectors creates numServers requestVectors
+// to retrieve data at the specified bucket
+func (c *Client) GenerateRequestVectors(bucket uint64, numServers uint64, numBuckets uint64) ([][]byte, error) {
+	if numServers < 2 {
+		c.log.Error.Printf("GenerateRequestVectors called with too few servers=%v", numServers)
+		return nil, fmt.Errorf("numServers=%v must be >1", numServers)
+	}
+	if bucket >= numBuckets {
+		c.log.Error.Printf("GenerateRequestVectors called with invalid bucket=%v, numBuckets=%v", bucket, numBuckets)
+		return nil, fmt.Errorf("bucket=%v must be <numBuckets=%v", bucket, numBuckets)
+	}
+
+	req := make([][]byte, numServers)
```

Comment:
Yes probably. Can I do this at a future PR when we're ready to switch to the new server implementation?